src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -286,8 +341,29 @@ def check_align_timestamps_internal(self):
                 )
             ]
 
+    def check_interface_align_starting_time_shifts_timestamps(self):
+        fresh_interface = self.data_interface_cls(**self.test_kwargs)
+        all_unaligned_timestamps = fresh_interface.get_timestamps()
+
+        starting_time = 1.23
+        fresh_interface.align_starting_time(starting_time=starting_time)
+
+        if fresh_interface._number_of_segments == 1:
+            aligned_timestamps = fresh_interface.get_timestamps()
+            expected_timestamps = all_unaligned_timestamps + starting_time
+            assert_array_equal(x=aligned_timestamps, y=expected_timestamps)
+        else:
+            all_aligned_timestamps = fresh_interface.get_timestamps()
+            all_expected_timestamps = [
+                unaligned_timestamps + starting_time for unaligned_timestamps in all_unaligned_timestamps
+            ]
+            [
```

Comment:
How does having assertions inside list comprehensions plays with the error trace? 

I feel that this is a bad idea from a debugging standpoint as I would not be able to stop in the specific case that failed if that comes to happen.

Reply:
Error gets thrown at same place and time as a normal for loop

Not sure what you mean about stopping; I assume that's some kind of IDE injection feature?

Still not sure why it was structured that way, no harm in making it a normal for loop

Reply:
Fixed

Reply:
I mean normal standard library debugger (pdb) tracepoint setting like:
```
    pdb.set_trace()  # This is where you set the debugger breakpoint
```

How do you debug when you need to jump to a certain line and inspect values?